### PR TITLE
Increase size of ride_id_t to 16 bits

### PR DESCRIPTION
**Pull request by Gymnasiast**
This does not yet modify the RIDE_ID_NULL value, which is planned to be addressed later.

### CODE REVIEW

File: src/openrct2/ride/RideTypes.h [hunk 1]
```diff
@@ -10,9 +10,10 @@
 #pragma once
 
 #include <cstdint>
+#include <limits>
 
-typedef uint8_t ride_id_t;
-typedef uint16_t ride_idnew_t; // Temporary, old one can be removed after switching save format.
+typedef uint16_t ride_id_t;
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/RideTypes.h]: Now's as good a time as any to switch to C++ style `using` instead of C style `typedef`.

Comment (Gymnasiast) [reply src/openrct2/ride/RideTypes.h]:
```suggestion
using ride_id_t = uint16_t;
```

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -2682,15 +2682,12 @@ static void vehicle_update_waiting_to_depart(Vehicle* vehicle)
...
 }
 
-#pragma pack(push, 1)
 struct rct_synchronised_vehicle
 {
     ride_id_t ride_id;
     uint8_t station_id;
     uint16_t vehicle_id;
 };
-assert_struct_size(rct_synchronised_vehicle, 4);
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp]: Why removing instead of updating? Just wondering

File: src/openrct2/world/TileElement.h [hunk 3]
```diff
@@ -189,7 +189,7 @@ struct PathElement : TileElementBase
     union
     {
         uint8_t additionStatus; // 7
-        ride_id_t rideIndex;
+        uint8_t rideIndex;
```
Comment (tupaschoal) [inline src/openrct2/world/TileElement.h]: This looks weird, it's the only place you let go of the `ride_id_t` alias, even though. Why?

Comment (IntelOrca) [reply src/openrct2/world/TileElement.h]: Changes to save import / export are needed before we can change the tile element structs, so the ride id in tile elements has to remain 8-bit until we make those further changes.

File: src/openrct2/ride/Vehicle.cpp (hunk 2 above)
Comment (IntelOrca) [reply src/openrct2/ride/Vehicle.cpp]: They were originally used to ensure we kept size compatibility with vanilla when interoping. We don't introp anymore and this struct is not used in the save file, or in networking so we can let the compiler adjust the struct however it likes.

**Pull request merged**